src/Metadata/HttpOperation.php

```diff
@@ -203,6 +207,13 @@ public function __construct(
         array|Parameters|null $parameters = null,
         array $extraProperties = [],
     ) {
+        if (null !== $this->errors) {
+            foreach ($this->errors as $error) {
+                if (!(new $error()) instanceof ProblemExceptionInterface) {
```

Comment:
let's avoid logic inside constructors especially on attributes, if a check is needed it should happen inside metadata factories.